### ISSUE TITLE
session was not delete from persistent storage.

### ISSUE THREAD

**Issue by xieyu** [bbangert/beaker#68]
def delete(self):
        """Deletes the session from the persistent storage, and sends
        an expired cookie out"""
        if self.use_cookies:
            self._delete_cookie()
        self.clear()

**Comment by pszalanski** [comment]
Yes, I ran into the same issue. I have a Redis beaker backend and the sessions keep piling up. `delete()` should have a statement like 
```python
del self.namespace['session']
```

**Comment by schlenk** [comment]
The clear method in _ConfigurableSession() assumes that a clear() on its superclass (dict) would call __delitem__, but it does not. So the session stays in persistent storage. This is insidious when using regenerate_id() as the old ID stays valid!
